sbncode/SBNEventWeight/Calculators/BNBFlux/FluxCalcPrep.cxx

```diff
@@ -0,0 +1,352 @@
+              fParameterSet.Sample(engine);//load it 2*<number_of_multisims> times
+            }
+  
+          }else {
+            throw cet::exception(__PRETTY_FUNCTION__) << GetName() << ": "
+              <<" calculator "+CalcType + "is invalid"
+              <<std::endl;
+          }
+
+
+        }//end of special Hadron calculator configurations
+
+      } else  validC = false; //the calculator name is way too off.
+      std::cout<<"SBNEventWeight : finish configuration."<<std::endl;
+    }//End of Configure() function
+
+
+    std::vector<float> FluxWeightCalc::GetWeight(art::Event& e, size_t inu) {
+//      std::cout<<"SBNEventWeight : getweight for the "<<inu<<" th particles of an event"<< std::endl;
+      //MCFlux & MCTruth
+      art::Handle< std::vector<simb::MCFlux> > mcFluxHandle;
+      e.getByLabel(fGenieModuleLabel,mcFluxHandle);
+      std::vector<simb::MCFlux> const& fluxlist = *mcFluxHandle;
+      //or do the above 3 lines in one line
+      auto const& mclist = *e.getValidHandle<std::vector<simb::MCTruth>>(fGenieModuleLabel);
+
+      // If mo neutrinos in this event, gives 0 weight;
+      int NUni = fParameterSet.fNuniverses;
+      std::vector<float> weights;//( mclist.size(), 0);
+      if(mclist.size() == 0){ 
+        std::cout<<"SBNEventWeight Flux: EMPTY WEIGHTS"<<std::endl;
+        return weights;
+      }
+
+      //Iterate through each neutrino in the event
+      if( CalcType == "Unisim"){//Unisim Calculator
+        weights.resize(NUni);
+        //Unisim specific
+        //containers for the parent and neutrino type information
+        int ptype = std::numeric_limits<int>::max(); 
+        int ntype = std::numeric_limits<int>::max();
+
+        // Discover the neutrino parent type
+        //     This contains the neutrino's parentage information
+        if (      fluxlist[inu].fptype==13  || fluxlist[inu].fptype==-13  ) ptype = 0;//mu
+        else if ( fluxlist[inu].fptype==211 || fluxlist[inu].fptype==-211 ) ptype = 1;//pi
+        else if ( fluxlist[inu].fptype==130                               ) ptype = 2;//K0
+        else if ( fluxlist[inu].fptype==321 || fluxlist[inu].fptype==-321 ) ptype = 3;//K
+        else {
+          throw cet::exception(__FUNCTION__) << GetName()<<"::Unknown ptype "<<fluxlist[0].fptype<< std::endl;
+        }
+
+        // Discover the neutrino type
+        //     This contains the neutrino's flavor information
+        if (      fluxlist[inu].fntype== 12  ) ntype=0;//nue
+        else if ( fluxlist[inu].fntype==-12  ) ntype=1;//nuebar
+        else if ( fluxlist[inu].fntype== 14  ) ntype=2;//numu
+        else if ( fluxlist[inu].fntype==-14  ) ntype=3;//numubar
+        else {
+          throw cet::exception(__FUNCTION__) << GetName()<<"::Unknown ntype "<<fluxlist[inu].fntype<< std::endl;
+        }
+
+        // Collect neutrino energy
+        double enu= mclist[inu].GetNeutrino().Nu().E();
```

Comment:
Instead of using MCTruth for this could you use:

`double enu = fluxlist[inu].fnenergy`

Then you could drop all references to `MCTruth`

Reply:
Sounds good! I just did not check it yet, but yes, if this works!

Reply:
It is `fluxlist[inu].fnenergyn` and I have applied the changes.